sdk/formrecognizer/ai-form-recognizer/src/lro/analyze.ts

```diff
@@ -0,0 +1,242 @@
+   */
+  confidence: number;
+}
+
+/**
+ * Transform a REST-level Document response object into the more strongly-typed AnalyzedDocument.
+ *
+ * @internal
+ *
+ * @param document - a REST-level document response object
+ * @returns an AnalyzedDocument (which has had its fields mapped to stronger DocumentField types)
+ */
+export function toAnalyzedDocumentFromGenerated(document: GeneratedDocument): AnalyzedDocument {
+  return {
+    ...document,
+    fields: toAnalyzedDocumentFieldsFromGenerated(document.fields ?? {}),
+  };
+}
+
+/**
+ * The result of an analysis operation. The type of the
+ */
+export interface AnalyzeResult<Document = AnalyzedDocument> {
+  /**
+   * The service API version used to produce this result.
+   *
+   * Example: "2020-09-30-preview"
+   */
+  apiVersion: FormRecognizerApiVersion;
+
+  /**
+   * The unique ID of the model that was used to produce this result.
+   */
+  modelId: string;
+
+  /**
+   * A string representation of all textual and visual elements in the input, concatenated by reading order (the order
+   * in which the service "reads" or extracts the textual anc visual content from the document).
+   */
+  content: string;
+
+  /** Analyzed pages. */
+  pages: DocumentPage[];
+
+  /**
+   * Extracted tables.
+   */
+  tables: DocumentTable[];
+
+  /**
+   * Extracted key-value pairs.
+   */
+  keyValuePairs: DocumentKeyValuePair[];
+
+  /**
+   * Extracted entities.
+   */
+  entities: DocumentEntity[];
+
+  /**
+   * Extracted font styles.
+   */
+  styles: DocumentStyle[];
+
+  /**
+   * Extracted documents (instances of any of the model's document types and corresponding field schemas) that were
+   * extracted from the input.
+   */
+  documents: Document[];
+}
+
+/**
+ * The state of an analysis operation, which will eventually produce the result type that corresponds to the model.
+ */
+export interface DocumentAnalysisPollOperationState<Result = AnalyzeResult<AnalyzedDocument>>
+  extends PollOperationState<Result> {
+  /**
+   * The status of the operation. One of:
+   *
+   * - "notStarted"
+   * - "running"
+   * - "succeeded"
+   * - "failed"
+   */
+  status: AnalyzeOperationStatus;
+
+  /**
+   * The model ID that the analysis operation will use to produce the result.
+   */
+  modelId: string;
+
+  /**
+   * The unique ID of this analysis operation.
+   */
+  operationId: string;
+}
+
+/**
+ * A long-running operation (poller) that tracks the state of an analysis operation, eventually producing the result
+ * type that corresponds to the model.
+ */
+export type AnalysisPoller<Result = AnalyzeResult<AnalyzedDocument>> = PollerLike<
+  DocumentAnalysisPollOperationState<Result>,
+  Result
+>;
+
+const operationLocationRegex = /\/documentModels\/([a-zA-Z0-9-]+)\/analyzeResults\/([a-z0-9-]+)/;
+
+/**
+ * Extract a model ID and analysis operation ID from an operationLocation URL.
+ * @internal
+ */
+export function parseOperationLocation(url: string | undefined): [string, string] {
```

Comment:
I am not a fan of parsing URLs like this. Did you consider hitting the full URL directly using `sendRequest`?

Reply:
I have some reservations about doing that. If a transform is applied to the swagger then I don`t think I would get the transforms/mappings if I send the request manually using the pipeline, and I would also have to add the options to the request myself.

Reply:
You can use the generated and transformed mapper. Also, another advantage of hitting the original URL is that it could have query parameters so they will be naturally preserved without having to parse them too and convert them to SDK options.

Reply:
I will add an issue for beta.2 to switch to this workflow.